trapsender/trap_sender.go

```diff
@@ -120,9 +121,25 @@ func (trapSender TrapSender) sendTraps(connectionArguments snmpgo.SNMPArguments,
 		snmp.Close()
 	}()
 
-	uptime, _ := host.Uptime()
-	if uptime > math.MaxInt32 {
-		uptime = 0
+	var uptime uint64
+	if trapSender.configuration.SNMPEngineStartTime == "" {
+		uptime, err = host.Uptime()
+		if err != nil {
+			trapSender.logger.Error("error while getting host uptime", "err", err.Error())
+			telemetry.SNMPTrapTotal.WithLabelValues(distinationForMetrics, "failure").Add(float64(len(traps)))
+			return err
+		}
+		if uptime > math.MaxInt32 {
+			uptime = 0
+		}
+	} else {
+		timestamp, err := strconv.Atoi(trapSender.configuration.SNMPEngineStartTime)
```

Comment:
Can you move this in the configuration parsing part?
Every error that can be handled at start time is preferred so that we 
don't have to wait for the first trap to be sent to notice there is a problem.
We can even handle the default value there, so that there is no special case to handle in the trap sender.